partiql-eval/src/eval.rs

```diff
@@ -1,12 +1,138 @@
-use crate::env::basic::MapBindings;
-use crate::env::Bindings;
-use partiql_value::Value::{Boolean, Missing, Null};
-use partiql_value::{Bag, BindingsName, Tuple, Value};
+use itertools::Itertools;
 use std::cell::RefCell;
 use std::collections::{HashMap, HashSet};
 use std::fmt::Debug;
 use std::rc::Rc;
 
+use petgraph::algo::toposort;
+use petgraph::prelude::StableGraph;
+use petgraph::{Directed, Incoming, Outgoing};
+
+use partiql_value::Value::{Boolean, Missing, Null};
+use partiql_value::{partiql_bag, Bag, BindingsName, Tuple, Value};
+
+use crate::env::basic::MapBindings;
+use crate::env::Bindings;
+
+#[derive(Debug)]
+pub struct EvalPlan(pub StableGraph<Box<dyn DagEvaluable>, (), Directed>);
+
+impl Default for EvalPlan {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl EvalPlan {
+    fn new() -> Self {
+        EvalPlan(StableGraph::<Box<dyn DagEvaluable>, (), Directed>::new())
+    }
+}
+
+pub type EvalResult = Result<Evaluated, EvalErr>;
+
+pub struct Evaluated {
+    pub result: Option<Value>,
+}
+
+pub struct EvalErr {
+    pub errors: Vec<EvaluationError>,
```

Comment:
To be useful, this will need, at a minimum to implement `std::error::Error`. Probably best to go with something like thiserror: https://docs.rs/thiserror/latest/thiserror/

No need to block this PR on the above, though.